clang/utils/TableGen/ClangDiagnosticsEmitter.cpp

```diff
@@ -1419,16 +1412,14 @@ void clang::EmitClangDiagsDefs(const RecordKeeper &Records, raw_ostream &OS,
   InferPedantic inferPedantic(DGParentMap, Diags, DiagGroups, DiagsInGroup);
   inferPedantic.compute(&DiagsInPedantic, (RecordVec*)nullptr);
 
-  for (unsigned i = 0, e = Diags.size(); i != e; ++i) {
-    const Record &R = *Diags[i];
-
+  for (const Record *RP : Diags) {
+    const Record &R = *RP;
```

Comment:
It's not ideal to have both `RP` and `R`.  You might want to clean this up in a follow-up patch or something.  Not a blocker for this patch.

Reply:
I can clean this up in this PR. I was looking into whether LLVM has any iterator adapters for this earlier and could not find one. But looking again, it seems `make_pointee_range` does what we want, so let me try it

Reply:
Done.